web/modules/custom/joinup_event/config/install/field.field.node.event.field_event_coordinates.yml

```diff
@@ -13,15 +13,15 @@ third_party_settings:
     weight: 0
     hidden: false
     disabled: false
+    providers:
+      - webtools_geocoding
```

Comment:
Just an observation and a subject of meditation: As `webtools_geocoding` is part of the field instance 3rd-party-settings, means that this field instance depends on `webtools_geocoding` config entity. As an effect, it should be listed in `field.field.node.event.field_event_coordinates.yml` under `dependencies.config`. We can do it manually, by editing the YAML file but, unfortunately, there's no API to do it in the `geocoder_field` (which is the module that adds the 3rd-party-setting). I've been discussed this, on Slack, with Alex Pott and he confirmed that there's no API mechanism that can hook into the "calculate dependencies" process and add/remove dependencies. I also suggested a future API expansion by adding a 4th parameter to `::setThirdPartySetting($module, $key, $value, $dependencies)`. In our case this would allow adding also dependencies along with the 3rd-party-settings. These are only thoughts around a configuration dependency inconsistency. But at geocoder level we should handle this in https://www.drupal.org/project/geocoder/issues/3016710.

Reply:
> I've been discussed this, on Slack, with Alex Pott and he confirmed that there's no API mechanism that can hook into the "calculate dependencies" process and add/remove dependencies. 

I had a look in core at the third party settings implementation and couldn't find a way to set the config dependencies from there. Thanks for confirming this with Alex. Shall we make a core issue about this? This should be part of the API.

Reply:
I added https://www.drupal.org/project/drupal/issues/3016895